### ISSUE TITLE
is_valid_auth_cookies() continually False, had to add values.get('SPOIDCRL')

### ISSUE THREAD

**Issue by nefariousOne** [vgrem/office365-rest-python-client#306]
I was trying to use this module to authenticate against our SPO tenant but would keep getting "An error occurred while retrieving auth cookies from ...". I tried with ClientContext vs UserContext which worked fine but anytime I tried to pass user credentials with UserContext it failed. 

After some investigation it was due to is_valid_auth_cookies() looking explicitly for "FedAuth" which our SSO provider does not pass, instead I had to add a match for "SPOIDCRL" as mentioned in this article https://techcommunity.microsoft.com/t5/microsoft-sharepoint-blog/sharepoint-online-active-authentication/ba-p/510052.

By replacing 
`def is_valid_auth_cookies(values):
    return any(values) and values.get('FedAuth', None) is not None`
with
`def is_valid_auth_cookies(values):
    return any(values) and (values.get('FedAuth', None) is not None or (values.get('SPOIDCRL') is not None))`

I hope this helps someone else out or, if I missed something, any guidance is always welcomed but at this point I can successfully connect to SPO using this module and pass the cookie along with subsequent requests.

**Comment by joshyu** [comment]
Yes, I also met with this issue. always failed when checking the session cookie

**Comment by vgrem** [comment]
Closing - the SAML/IDCRL cookie flow has been removed following Microsoft's permanent retirement of the IDCRL protocol on May 1, 2026 ([MC1184649](https://devblogs.microsoft.com/microsoft365dev/migrating-from-idcrl-authentication-to-modern-authentication-in-sharepoint/)).
Modern replacements are available:

```python
# Interactive (MFA-compatible)
ctx = ClientContext(site_url).with_interactive(tenant, client_id)

# MSAL ROPC (non-interactive, no MFA)
ctx = ClientContext(site_url).with_username_and_password(tenant, client_id, username, password)

# App-only (recommended for automation)
ctx = ClientContext(site_url).with_client_certificate(tenant, client_id, thumbprint, cert_path)
```


Install from master if you need this before the next PyPI release: pip install `git+https://github.com/vgrem/office365-rest-python-client.git`
